service/service_windows.go

```diff
@@ -0,0 +1,146 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build windows
+
+package service
+
+import (
+	"fmt"
+	"syscall"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap/zapcore"
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/eventlog"
+)
+
+type WindowsService struct {
+	params Parameters
+	app    *Application
+}
+
+func NewWindowsService(params Parameters) *WindowsService {
+	return &WindowsService{params: params}
+}
+
+func (s *WindowsService) Execute(args []string, requests <-chan svc.ChangeRequest, changes chan<- svc.Status) (bool, uint32) {
```

Comment:
I suppose this function implements a function from the `svc` package, but would be nice to leave comments describing the returned `bool` and `uint32` values, e.g. `(bool /* svcSpecificEC */, int32 /* exitCode */)`.

You may also name returned values via naked return mechanism, but I'm personally not a fan of that Go feature since it implicitly declares variables. See discussion https://github.com/golang/go/issues/21291 if interested.

Or at least leave a link to https://godoc.org/golang.org/x/sys/windows/svc#Handler in the comment.

Reply:
👍 - left a comment to the docs and named the return variables